test/testcmdlineparser.cpp

```diff
@@ -104,6 +104,19 @@ class TestCmdlineParser : public TestFixture {
         // TODO: verify that the redirect output is empty
     }
 
+    // add specialization so the enums can be compared without a cast
+    template<typename T, typename U>
```

Comment:
I had to add the general version because I cannot create a specialization for the template in the parent class.